cpp/include/cuvs/neighbors/all_neighbors.hpp

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuvs/neighbors/ivf_pq.hpp>
+#include <cuvs/neighbors/nn_descent.hpp>
+
+#include <variant>
+
+namespace cuvs::neighbors::all_neighbors {
+
+/**
+ * @brief Parameters used to build an all-neighbors knn graph (find nearest neighbors for all the
+ * training vectors)
+ */
+namespace graph_build_params {
```

Comment:
Are these copied over from CAGRA? Are they much different than CAGRA today? Any opportunity to reuse here would be great (by maybe putting this in `common.h`.